challenges/security/application-security.md

```diff
@@ -0,0 +1,270 @@
+* It helps you get a feel for what it would be like to work at Teleport, as this
+  exercise aims to simulate our day-as-usual and expose you to the type of work
+  and communication you can expect if you join Teleport.
+* We aim to keep the process light. While there are time limits, it is to keep
+  exercises from consuming undue amounts of your time, and not to test
+  performance under pressure.
+
+We believe this technique is not only better, but also is more fun compared to
+whiteboard/quiz interviews so common in the industry.
+
+[Some of the best teams use coding challenges.](https://sockpuppet.org/blog/2015/03/06/the-hiring-post/)
+
+We appreciate your time and are looking forward to the interview.
+
+# Leveling
+There are 6 engineering levels at Teleport. It's possible to earn level 4-5 via
+this interview process.
+
+We are not currently hiring L1-L3 security engineers and level 6 is only
+available via internal promotion. See our
+[security engineering levels](https://github.com/gravitational/careers/blob/main/levels/security.pdf)
+for more details.
+
+We expect different performance at different engineering levels. These
+expectations are explicitly documented below.
+
+# Part 1: Hiring Manager Kick Off
+The hiring manager will discuss your background and approach to security with
+you.
+
+
+# Part 2: Peer Review
+You will be given a GitHub repo containing a PR that includes a
+[RFD](https://github.com/gravitational/teleport/blob/master/rfd/0000-rfds.md)
+and architecture diagram for a service similar to docker.
+
+You will have 90 minutes of asynchronous time to read through the RFD and
+provide written PR feedback in GitHub.
+
+Please be sure to cover security concerns, though you're welcome to provide
+input in any area where you see room for improvement.
+
+## Level 4
+Review the service as written, assuming it will run on a single server.
+
+## Level 5
+The service will run in multiple geographies with unreliable network connections
+between them. Please address the security and business tradeoffs between
+authentication and authorization consistency architectures.
+
+# Part 3: Break
+Grab a bite of food and some water. Take a breather before we continue to the
+next section.
+
+# Part 4: Development
+You will be given a prototype go implementation of the RFD reviewed in
+part 1. You will have 3 hours to develop and submit a single PR implementing
+an authorization and authentication scheme.
+
+This is asynchronous. You'll develop on your machine and you're welcome to
+use whatever tools you're comfortable with to develop the changes. We'll be
+available in the interview slack channel to answer any questions.
+
+We will review your PR, including compiling and run the program and testing
+it behavior. Make sure to write good commit messages, as well as a helpful
+PR description.
+
+This prototype is expected to run on only a single machine. The distributed
+state considerations of Part 2 L5 need not be considered.
```

Comment:
Clarify what L5 is here.

Reply:
Changed to "Level 5". Let me know if further clarification is needed.